tox_core/src/onion/client/mod.rs

```diff
@@ -533,16 +533,16 @@ impl OnionClient {
     }
 
     /// Handle DHT `PublicKey` announce from both onion and DHT.
-    pub fn handle_dht_pk_announce(&self, friend_pk: PublicKey, dht_pk_announce: DhtPkAnnouncePayload) -> impl Future<Output = Result<(), HandleDhtPkAnnounceError>> + Send {
-        let mut state = self.state.lock();
+    pub async fn handle_dht_pk_announce(&self, friend_pk: PublicKey, dht_pk_announce: DhtPkAnnouncePayload) -> Result<(), HandleDhtPkAnnounceError> {
+        let mut state = self.state.lock().clone();
```

Comment:
Why does it become clone? This way it's not atomic update anymore.

Reply:
Tried using `Copy`, but compiler complains that the lifetime extends past the `await` call below.

I agree, this is hacky. Can you see an obviously better way to do this? Will keep hacking on it for a better solution.

Marking this PR as [WIP] until then.

Reply:
@kurnevsky after wrestling with this for a few more hours, the best thing I could half-way get working ended up looking very similar to what is there now (with `impl Future<Output...>` return).

Am sure there is a way to accomplish the refactor, but maybe it takes a total refactor of all the async components.

Will re-attempt the refactor when more familiar with the library, and with async/await in general. Apologies if I wasted any of your time.